hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -134,8 +136,11 @@ public MapContainer(final String name, final Config config, final MapServiceCont
         initWanReplication(nodeEngine);
         this.extractors = new Extractors(mapConfig.getMapAttributeConfigs(), config.getClassLoader());
         if (shouldUseGlobalIndex(mapConfig)) {
+            IndexCopyBehavior copyBehavior = mapServiceContext.getProperties().getEnum(QUERY_INDEX_RESULT_COPY,
```

Comment:
Cant we get this `QUERY_INDEX_RESULT_COPY` property directly inside `BaseIndexStore`? As far as i see, it doesn't seem a per map setting.

Reply:
I didn't want to leak the HazelcastProperties to the Indexes, it makes the testing so much harder.

Reply:
I see. But `MapServiceContextImpl` seems a better place because `MapContainer` is for per map stuff.

Reply:
Yes, that's a helpful hint. I encapsulated the reading of the property in the MapServiceContextImpl.